app/assets/v2/js/pages/bounty_details.js

```diff
@@ -529,7 +532,30 @@ var attach_work_actions = function() {
   $('body').delegate('a[href="/interested"], a[href="/uninterested"], a[href="/extend-deadlines"]', 'click', function(e) {
     e.preventDefault();
     if ($(this).attr('href') == '/interested') {
-      show_interest_modal.call(this);
+      if (currentLoggedInUser.length > 0 && reservedForUsername.length > 0) {
```

Comment:
This ensures if a user clicks on start work (which is reserved for someone else ), he'd see a modal with the waiting message ( As an end user I would be frustrated to click a bounty and to only know it's reserved for someone else )

Should we hide the start work button for other contributors (aka do a check on the `details.html`) until the issue is open to everyone? 

cc @PixelantDesign @mbeacom @SaptakS